flipt-engine-ffi/src/lib.rs

```diff
@@ -212,7 +212,7 @@ pub unsafe extern "C" fn initialize_engine(
 
     let engine_opts_bytes = CStr::from_ptr(opts).to_bytes();
     let bytes_str_repr = std::str::from_utf8(engine_opts_bytes).unwrap();
-    let engine_opts: EngineOpts = serde_json::from_str(bytes_str_repr).unwrap_or_default();
+    let engine_opts: EngineOpts = serde_json::from_str(bytes_str_repr).unwrap();
```

Comment:
i guess we panic above though as there are several unwrap calls above this. so ignore me. I've actually been bitten by this before when doing the C# client, where it uses the defaults instead of failing fast so this is probably better they way you have it

Reply:
ah yes sorry this was from debugging